Added files for MOT17

NishantSinghhhhh opened the pull request:
### What type of PR is this?
/kind documentation
/kind feature
/kind cleanup

### What this PR does / why we need it:

This PR completely revamps the `examples/MOT17/multiedge_inference_bench` example to elevate it to a "gold standard" of runnable documentation. It directly addresses the problems of poor usability, manual setup, and lack of clarity outlined in issue #231 
The key improvements include:

NishantSinghhhhh committed:
Added files for MOT17

Signed-off-by: NishantSinghhhhh <nishantsingh_230137@aitpune.edu.in>

changes

Signed-off-by: NishantSinghhhhh <nishantsingh_230137@aitpune.edu.in>

changes

Signed-off-by: NishantSinghhhhh <nishantsingh_230137@aitpune.edu.in>

changes

Signed-off-by: NishantSinghhhhh <nishantsingh_230137@aitpune.edu.in>

Made MOT 17 work

Signed-off-by: NishantSinghhhhh <nishantsingh_230137@aitpune.edu.in>

Added code

Signed-off-by: NishantSinghhhhh <nishantsingh_230137@aitpune.edu.in>

NishantSinghhhhh commented:
@MooreZheng,

Could you please review this PR when you have a moment? I've aimed to overhaul the MOT17 benchmark example based on the new quality standards. Let me know if any changes are needed.

On a separate note, I was hoping to inquire if the Ianvs project plans to participate in the LFX Mentorship program for Term 3 this year.

MooreZheng reviewed (changes requested):
The current challenge and solution are not super clear for reviewers. A few suggestions:  
1) Establish a issue to describe what is wrong with the current MOT17 example.
2) Link this PR to the corresponding issue and point out how this PR solve the problems.

NishantSinghhhhh commented:
@MooreZheng , I have 1 question 

for MOT17 to run we need data and models running, so for that should I be adding everything in the repository or there is a better way of  doing it ?

NishantSinghhhhh commented:
@MooreZheng , I have 1 question 

for MOT17 to run we need data and models running, so for that should I be adding everything in the repository or there is a better way of  doing it ?

NishantSinghhhhh commented:
[Screencast from 2025-08-02 23-05-20.webm](https://github.com/user-attachments/assets/3d1b2588-3b6c-48b7-8036-b38a0d605caa)

@MooreZheng I have made the MOT work, I have made a few changes in the structure of MOT17, 

- I have added a workspace folder in which all the data to be processed is present and that in csv format
- report that is generated comes in reports folder inside pedestrian_tracking
- I had to change paths written in reid_job.yaml & tracking_job.yaml

Can you review my PR and tell me if any changes are required, I have attached a video too, in which the report is successfully generated

MooreZheng commented:
> @MooreZheng sir, can you please review the changes

Our KubeEdge-ianvs reviewer @hsj576 might also want to take a look at the PR first

hsj576 reviewed (changes requested):
Overall, this PR looks fine to me, but the author need to merge the commits into a single commit.

hsj576 commented:
/lgtm

MooreZheng commented on examples/MOT17/multiedge_inference_bench/workspace/reid_job/rank/all_rank.csv:
```diff
@@ -0,0 +1,2 @@
+time,paradigm,algorithm,basemodel,batch_size,mAP,rank_1,rank_2,rank_5,cmc
```
Do not upload CSV to the KubeEdge Ianvs GitHub repository. The author can use Kaggle to keep the ianvs version of MOT17, the ranking CSV result, and the generated report, with a link provided in the readme in the ianvs repo. As for Kaggle, add kubeedge-ianvs as an admin for the dataset so members can find it easily. See [Kaggle link](https://www.kaggle.com/kubeedgeianvs/datasets)

MooreZheng commented on examples/MOT17/multiedge_inference_bench/workspace/tracking_job/rank/all_rank.csv:
```diff
@@ -0,0 +1,2 @@
+time,paradigm,algorithm,basemodel,batch_size,precision,recall,f1_score,mota,motp,idf1
```
Do not upload CSV to the KubeEdge Ianvs GitHub repository.  See comment above

MooreZheng commented on examples/MOT17/multiedge_inference_bench/pedestrian_tracking/reports/20250802232515.pdf:
Do not upload a PDF to the KubeEdge Ianvs GitHub repository.  The author can use Kaggle to keep the ianvs version of MOT17, the ranking CSV result, and the generated report, with a link provided in the readme in the ianvs repo. As for Kaggle, add kubeedge-ianvs as an admin for the dataset so members can find it easily. See [Kaggle link](https://www.kaggle.com/kubeedgeianvs/datasets)

MooreZheng reviewed (changes requested):
Overall, it looks fine. A few suggestions: 
1) Use Python logger instead of print.
2) Use Kaggle, rather than the Ianvs Github repo, to store the PDF and CSV so that the repo will not get too large for developers.

NishantSinghhhhh commented:
@MooreZheng sir , i will make the changes and commit

NishantSinghhhhh committed:
Added dataset and report to kaggle, removed them from codebase

Signed-off-by: NishantSinghhhhh <nishantsingh_230137@aitpune.edu.in>

NishantSinghhhhh committed:
Added docs

Signed-off-by: NishantSinghhhhh <nishantsingh_230137@aitpune.edu.in>

NishantSinghhhhh commented:
@MooreZheng sir I have changed the readme can you tell me

<img width="1228" height="111" alt="image" src="https://github.com/user-attachments/assets/cbf6833c-9718-4ac5-aa56-42e4c4fa2f7f" />

are these the correct headings I have added, and id yes then what should I be adding in the docs for "Pedestrian Tracking on Multiple Object Tracking"

NishantSinghhhhh commented:
@MooreZheng sir, I have added the changes you have asked me to, can you please me, what changes should I do in the docs, so that MOT17 will be complete

NishantSinghhhhh committed:
Added docs

Signed-off-by: NishantSinghhhhh <nishantsingh_230137@aitpune.edu.in>

NishantSinghhhhh commented:
@MooreZheng sir I have added the docs in readme, can you please review the docs I have added and tell me if any changes are required
